cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -565,14 +565,28 @@ public static void updateStatus(Throwable error, KafkaConnector kafkaConnector2,
         }
         StatusUtils.setStatusConditionAndObservedGeneration(connector, status, error != null ? Future.failedFuture(error) : Future.succeededFuture());
         status.setConnectorStatus(statusResult);
-        status.setTasksMax(connector.getSpec().getTasksMax());
+
+        status.setTasksMax(getActualTaskCount(connector, statusResult));
 
         return maybeUpdateStatusCommon(connectorOperator, connector, reconciliation, status,
             (connector1, status1) -> {
                 return new KafkaConnectorBuilder(connector1).withStatus(status1).build();
             });
     }
 
+    protected int getActualTaskCount(KafkaConnector connector, Map<String, Object> statusResult)  {
```

Comment:
I wonder if it is worth a small doc comment explaining why this is the 'actual' value and why tasksMax is sometimes not set?